src/test/integration/utils.py

```diff
@@ -0,0 +1,61 @@
+import argparse
+import sys
+import requests
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-i", help="path to input file", dest="input_file")
+    namespace, args = parser.parse_known_args()
+    return namespace, sys.argv[:1] + args
+
+
+def set_local_vars(config):
+    global url
+    url_pieces = (config["hostname"], config["version"], config["api"])
+    url = url_joiner(url_pieces)
+    print("url: {}".format(url))
+
+    global session
+    session = requests.Session()
+    if config["use_basic_auth"]:
+        basic_auth = config["basic_auth"]
+        session.auth = (basic_auth["username"], basic_auth["password"])
+        session.verify = False
+    else:
+        session.headers = get_oauth2_headers(config)
+    print(session.auth)
```

Comment:
My mistake. I forgot to delete the print statements I was using while I was testing my URL builder and authentication. `None` is printed because `session.auth` is not set when using OAUTH2 like it is when basic_auth is used. Instead, the authentication for OAUTH2 is set in the headers. `FFF` is printed from the test suite. `unittest` will indicate the outcome of each test method by one of the following: `.` means success, `F` means fail, and `E` means error. Running with the `-v` flag will print out the name of each test method while it's executing which can be preferred when conveying what went wrong.